packages/primitives-core/src/types.ts

```diff
@@ -0,0 +1,10 @@
+type NamedStyles<T> = { [P in keyof T]: unknown }
+
+// This is based on the StyleSheet type from @types/react-native
+export interface AbstractStyleSheet {
+  create<T extends NamedStyles<T> | NamedStyles<any>>(
```

Comment:
Hmm, it's an interesting constraint. I can't say that I fully understand why it is written like this. I guess this will be OK or we'll just learn that it's not when converting `@emotion/native` and stuff 🤷‍♂️